react/src/components/CreateMapModal/CreateMapModal.tsx

```diff
@@ -0,0 +1,210 @@
+import React, { useState } from 'react';
+import {
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  FormGroup,
+  Label,
+  Input,
+} from 'reactstrap';
+import { Button } from '../../core-components';
+import styles from './CreateMapModal.module.css';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+import useCreateProject from '../../hooks/projects/useCreateProject';
+import useAuthenticatedUser from '../../hooks/user/useAuthenticatedUser';
+import { useNavigate } from 'react-router-dom';
+
+type CreateMapModalProps = {
+  isOpen: boolean;
+  toggle: () => void;
+};
+
+// Yup validation schema
+const validationSchema = Yup.object({
+  name: Yup.string().required('Name is required'),
+  description: Yup.string().required('Description is required'),
+  system_file: Yup.string()
+    .matches(
+      /^[A-Za-z0-9-_]+$/,
+      'Only letters, numbers, hyphens, and underscores are allowed'
+    )
+    .required('Custom file name is required'),
+});
+
+const CreateMapModal = ({ isOpen, toggle }: CreateMapModalProps) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const { data: userData } = useAuthenticatedUser();
+  const { mutate: createProject, isLoading: isCreatingProject } =
+    useCreateProject();
+  const navigate = useNavigate();
+
+  const handleCreateProject = (projectData, actions) => {
```

Comment:
```suggestion
  const handleCreateProject = (projectData: ProjectRequest, actions) => {
```
would be good to use type here.